File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using System.Drawing.Printing;
using DevExpress.XtraPrintingLinks;
using System.Drawing.Imaging;
using System.IO;

namespace PrintImageFrames {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            ReportAnImage(Application.StartupPath + @"\..\..\" + "Rotating_earth_(small).gif", checkBox1.Checked);
        }

        private void button2_Click(object sender, EventArgs e) {
            ReportAnImage(Application.StartupPath + @"\..\..\" + "Prices.tiff", checkBox1.Checked);
        }

        private void ReportAnImage(string filename, bool pageBreakAfterEachImage) {
            CompositeLink compositeLink = new CompositeLink(printingSystem1);
            Image image = Image.FromFile(filename);
            FrameDimension frameDimension = new FrameDimension(image.FrameDimensionsList[0]);
            int framesCount = image.GetFrameCount(frameDimension);

            for(int i = 0; i < framesCount; i++) {
                image.SelectActiveFrame(frameDimension, i);
                compositeLink.Links.Add(new ImageLink(new Bitmap(image), pageBreakAfterEachImage));
            }

            image.Dispose();

            compositeLink.PaperKind = PaperKind.A3;
            compositeLink.Landscape = true;

            compositeLink.ShowPreviewDialog();
        }

    }

    public class ImageLink : Link {

        private Image image;

        public Image Image {
            get { return image; }
            set { image = value; }
        }

        private bool addPageBreak;

        public bool AddPageBreak {
            get { return addPageBreak; }
            set { addPageBreak = value; }
        }

        public ImageLink()
            : this(null, null, false) {

        }

        public ImageLink(Image image, bool addPageBreak)
            : this(null, image, addPageBreak) {

        }

        public ImageLink(PrintingSystem ps, Image image, bool addPageBreak)
            : base(ps) {

            this.Image = image;
            this.AddPageBreak = addPageBreak;
        }

        protected override void CreateDetail(BrickGraphics graph) {
            if(Image != null) {
                // Add an image to a specific location.
                graph.DrawImage(Image, new Rectangle(0, 0, Image.Width, Image.Height),
                    BorderSide.None, Color.Transparent);

                if (AddPageBreak)
                    // Add a page break after the image.
                    graph.PrintingSystem.InsertPageBreak(Image.Height + 1);
            }
        }

    }

}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports System.Drawing.Printing
Imports DevExpress.XtraPrintingLinks
Imports System.Drawing.Imaging
Imports DevExpress.Drawing.Printing

Namespace PrintImageFrames

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            ReportAnImage(Application.StartupPath & "\..\..\" & "Rotating_earth_(small).gif", checkBox1.Checked)
        End Sub

        Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs)
            ReportAnImage(Application.StartupPath & "\..\..\" & "Prices.tiff", checkBox1.Checked)
        End Sub

        Private Sub ReportAnImage(ByVal filename As String, ByVal pageBreakAfterEachImage As Boolean)
            Dim compositeLink As CompositeLink = New CompositeLink(printingSystem1)
            Dim image As Image = Image.FromFile(filename)
            Dim frameDimension As FrameDimension = New FrameDimension(image.FrameDimensionsList(0))
            Dim framesCount As Integer = image.GetFrameCount(frameDimension)
            For i As Integer = 0 To framesCount - 1
                image.SelectActiveFrame(frameDimension, i)
                compositeLink.Links.Add(New ImageLink(New Bitmap(image), pageBreakAfterEachImage))
            Next

            image.Dispose()
            compositeLink.PaperKind = DXPaperKind.A3
            compositeLink.Landscape = True
            compositeLink.ShowPreviewDialog()
        End Sub
    End Class

    Public Class ImageLink
        Inherits Link

        Private imageField As Image

        Public Property Image As Image
            Get
                Return imageField
            End Get

            Set(ByVal value As Image)
                imageField = value
            End Set
        End Property

        Private addPageBreakField As Boolean

        Public Property AddPageBreak As Boolean
            Get
                Return addPageBreakField
            End Get

            Set(ByVal value As Boolean)
                addPageBreakField = value
            End Set
        End Property

        Public Sub New()
            Me.New(Nothing, Nothing, False)
        End Sub

        Public Sub New(ByVal image As Image, ByVal addPageBreak As Boolean)
            Me.New(Nothing, image, addPageBreak)
        End Sub

        Public Sub New(ByVal ps As PrintingSystem, ByVal image As Image, ByVal addPageBreak As Boolean)
            MyBase.New(ps)
            Me.Image = image
            Me.AddPageBreak = addPageBreak
        End Sub

        Protected Overrides Sub CreateDetail(ByVal graph As BrickGraphics)
            If Image IsNot Nothing Then
                ' Add an image to a specific location.
                graph.DrawImage(Image, New Rectangle(0, 0, Image.Width, Image.Height), BorderSide.None, Color.Transparent)
                ' Add a page break after the image.
                If AddPageBreak Then graph.PrintingSystem.InsertPageBreak(Image.Height + 1)
            End If
        End Sub
    End Class
End Namespace
